Sources/SourceKitLSP/Swift/CodeActions/ConvertStringConcatenationToStringInterpolation.swift

```diff
@@ -0,0 +1,139 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SwiftRefactor
+import SwiftSyntax
+
+struct ConvertStringConcatenationToStringInterpolation: SyntaxRefactoringProvider {
+  static func refactor(syntax: SequenceExprSyntax, in context: Void) -> SequenceExprSyntax? {
+    guard let (componentsOnly, commonPounds) = preflight(exprList: syntax.elements) else {
+      return nil
+    }
+
+    var ret: StringLiteralSegmentListSyntax = []
+    for component in componentsOnly {
+      if let stringLiteral = StringLiteralExprSyntax(component) {
+        var segments = stringLiteral.segments
+        if stringLiteral.openingPounds?.tokenKind != commonPounds?.tokenKind {
+          for i in segments.indices {
+            guard case var .expressionSegment(exprSegment) = segments[i] else {
+              continue
+            }
+            exprSegment.pounds = commonPounds
+            segments[i] = .expressionSegment(exprSegment)
+          }
+        }
+        ret += segments
+      } else {
+        ret.append(
+          .expressionSegment(
+            ExpressionSegmentSyntax(
+              pounds: commonPounds,
+              expressions: [
+                LabeledExprSyntax(expression: component.trimmed)
+              ]
+            )
+          )
+        )
+      }
+    }
+
+    return SequenceExprSyntax(elements: [
+      ExprSyntax(
+        StringLiteralExprSyntax(
+          openingPounds: commonPounds,
+          openingQuote: "\"",
+          segments: ret,
+          closingQuote: "\"",
+          closingPounds: commonPounds
+        )
+      )
+    ])
+  }
+
+  /// If `exprList` is a string concatenation, returns 1) all elements in `exprList` with concat operators stripped and 2) the longest pounds amongst all string literals,
+  /// otherwise returns nil.
+  ///
+  ///  `exprList` as a valid string concatenation must contain n >= 3 children where n is an odd number with a concat operator `+` separating every other child which must either be a string literal or a valid expression for string interpolation. `exprList` must also contain at least one string literal child.
+  ///
+  ///  For example,
+  ///
+  ///      "Hello " + aString + "\(1)World"
```

Comment:
Let’s use markdown to format these code snippets.

```suggestion
  /// ```swift
  /// "Hello " + aString + "\(1)World"
  /// ```
```

Same below